pkg/types/gcp/platform.go

```diff
@@ -30,3 +33,9 @@ type Platform struct {
 	// +optional
 	ComputeSubnet string `json:"computeSubnet,omitempty"`
 }
+
+//SetPublicZoneID sets public cloud DNS zone ID for gcp.Platform
+func (p *Platform) SetPublicZoneID(zoneID string) error {
```

Comment:
i'm not a big fan of these setters, the field itself is public so the consumers can modify/set it directly.

Reply:
I am not a fan either, was just following the existing convention ;-) I have removed the setter and replaced it with a direct assignment.